### PR TITLE
[TIR] Register CUDA WMMA tensor intrinsics

### PR DESCRIPTION
**Pull request by vinx13**
This PR registers CUDA wmma intrinsics. Currently, only 16x16x16 wmma intrinsics are supported. Input should be `float16`. Accumulator can be `float16` or `float32`. Fragment B can be transposed (column major).

cc @masahi @junrushao1994

### CODE REVIEW

**Comment by masahi** [comment]
How about using these intrins in `test_tir_schedule_tensorize.py` to replace ad-hoc ones?

**Comment by vinx13** [comment]
> How about using these intrins in `test_tir_schedule_tensorize.py` to replace existing ad-hoc ones?

The intrinsics in the test case is more like toy examples that are not actually runnable by hardware. To use these real intrinsics, we will also need to do cache read into wmma fragment. It seems that will make those test cases more complicated for unit test purpose.

File: python/tvm/tir/tensor_intrin/cuda.py [hunk 1]
```diff
@@ -482,3 +484,325 @@ def mma_store_impl(a: T.handle, c: T.handle) -> None:
...
+WMMA_SYNC_16x16x16_f16f16f32_INTRIN = "wmma_sync_16x16x16_f16f16f32"
+TensorIntrin.register(
+    WMMA_SYNC_16x16x16_f16f16f32_INTRIN,
+    *get_wmma_sync_intrin(16, 16, 16, "float16", "float32", False),
+)
+
+WMMA_SYNC_16x16x16_f16f16f32_TRANS_INTRIN = "wmma_sync_16x16x16_f16f16f32_trans"
+TensorIntrin.register(
+    WMMA_SYNC_16x16x16_f16f16f32_TRANS_INTRIN,
+    *get_wmma_sync_intrin(16, 16, 16, "float16", "float32", False),
```
Comment (masahi) [inline python/tvm/tir/tensor_intrin/cuda.py:745]: The last arg should be `True`?

File: python/tvm/tir/tensor_intrin/cuda.py [hunk 2]
```diff
@@ -482,3 +484,325 @@ def mma_store_impl(a: T.handle, c: T.handle) -> None:
...
+WMMA_SYNC_16x16x16_f16f16f16_INTRIN = "wmma_sync_16x16x16_f16f16f16"
+TensorIntrin.register(
+    WMMA_SYNC_16x16x16_f16f16f16_INTRIN,
+    *get_wmma_sync_intrin(16, 16, 16, "float16", "float16", False),
+)
+
+WMMA_SYNC_16x16x16_f16f16f16_TRANS_INTRIN = "wmma_sync_16x16x16_f16f16f16_trans"
+TensorIntrin.register(
+    WMMA_SYNC_16x16x16_f16f16f16_TRANS_INTRIN,
+    *get_wmma_sync_intrin(16, 16, 16, "float16", "float16", False),
```
Comment (masahi) [inline python/tvm/tir/tensor_intrin/cuda.py:757]: This one too.

File: python/tvm/tir/tensor_intrin/cuda.py (hunk 1 above)
Comment (vinx13) [reply python/tvm/tir/tensor_intrin/cuda.py]: Good catch

**Review by masahi — APPROVED** [review]

**Pull request merged**